App/components/Bible/Chapter.js

```diff
@@ -3,5 +3,5 @@ import React from 'react';
 import Verse from './Verse';
 
 export default function Chapter({ verses }) {
-	return verses?.map((verse, i) => <Verse key={i} number={i + 1} text={verse} />);
-}
\ No newline at end of file
+  return verses?.map((verse, i) => <Verse key={`${i} ${verse}`} number={i + 1} text={verse} />);
```

Comment:
I changed the key here because it is not recommended to use an array index as a key. Also I feel that we shouldn't be generating verse numbers but instead we should probably be passing verse numbers into the Verse component.

Reply:
> Also I feel that we shouldn't be generating verse numbers but instead we should probably be passing verse numbers into the Verse component.

We are passing verse numbers to the verse component - its the `number` prop

Reply:
The issue is that the verse number is being generated from the array index: `number={i + 1}`. I feel it would be best to pass in the verse number from the JSON file itself rather than passing in a generated number.

Reply:
I see what you are saying. It all comes down to the fact that I changed the translation json files to store book, chapter, and verses as nested arrays to save on file size.

We could play around with useBible mapping over the verses object and returning something like
```
[{ verse: 1, text: 'and it came to pass' }]
```

If you want to play around with it and see if you can come up with something better, check out useBible.js - thats where all that logic should reside